src/foam/dao/FixedSizeDAO.js

```diff
@@ -13,22 +13,38 @@ foam.CLASS({
 
   javaImports: [
     'foam.dao.Sink',
-    'org.apache.commons.collections.buffer.CircularFifoBuffer',
-    'org.apache.commons.collections.Buffer',
-    'org.apache.commons.collections.BufferUtils'
+    // why am I using a ReentractLock ?
+    'java.util.concurrent.locks.ReentrantLock'  
   ],
 
   properties: [
+    {
+      class: 'Int',
+      name: 'nextIndex'
+    },
     {
       class: 'Int',
       name: 'fixedArraySize',
       value: 10000
     },
+    {
+      class: 'Int',
+      name: 'internalArraySize',
```

Comment:
document the internal/external array and sizes and a bit about why the internal is created a bit larger.